contracts/StakeManager.sol

```diff
@@ -165,12 +172,15 @@ contract StakeManager is Ownable {
      * @dev Reverts when `_time` is bigger than `MAX_LOCKUP_PERIOD`
      * @dev Reverts when `_time + block.timestamp` is smaller than current lock time.
      */
-    function lock(uint256 _time) external onlyVault noMigration processEpoch {
+    function lock(uint256 _time) external onlyVault onlyInitialized(msg.sender) noMigration processEpoch {
         if (_time > MAX_LOCKUP_PERIOD) {
             revert StakeManager__InvalidLockupPeriod();
         }
         Account storage account = accounts[msg.sender];
         processAccount(account, currentEpoch);
+        if (account.lockUntil + _time < block.timestamp) {
+            revert StakeManager__InvalidLockTime();
```

Comment:
So one thing I've realised, is that I keep confusing lock**Until** with lock**time**.
The former is a threshold, the latter is a period.

If we treat `_time` as something that can only be **added** to the existing `lockUntil`, then maybe we can consider renaming `lock()` to `increaseLockTime()` ?

Not something that has to happen in this commit/PR, but maybe something to consider in the future.

I always thought `lock()` would allow me to set my **lockup** time (say, from 2 years, to 4 years)

Reply:
Yes, that can be renamed like that to avoid confusions on development and audits.

Anyway, the UX would have to solve that for the user, no matter how its named or how the interface behaves.